tests/test_create_flatfield.py

```diff
@@ -0,0 +1,174 @@
+    if not os.path.exists(file_dir):
+        os.mkdir(file_dir) 
+    filenames = glob.glob(os.path.join(data_dir, "med*.fits"))
+    data_set = data.Dataset(filenames)
+    # creating flatfield for neptune for band 1
+    planet='neptune'; band='1'
+    mocks.create_onsky_rasterscans(data_set,filedir=file_dir,planet='neptune',band='1',im_size=420,d=50, n_dith=3,numfiles=36,radius=54,snr=250,snr_constant=4.55)
+    
+    ####### create flat field 
+    flat_dataset=[]
+    flat_filenames = glob.glob(os.path.join(file_dir, "*.fits"))
+    flat_dataset_all = data.Dataset(flat_filenames)
+    for i in range(len(flat_dataset_all)):
+        target=flat_dataset_all[i].pri_hdr['TARGET']
+        filter=flat_dataset_all[i].pri_hdr['FILTER']
+        if planet==target and band==filter: 
+            flat_dataset.append(flat_dataset_all[i])
+    onskyflat_field = detector.create_onsky_flatfield(flat_dataset, planet='neptune',band='1',up_radius=55, im_size=420, N=3, rad_mask=1.26,  planet_rad=50, n_pix=165, n_pad=302)
+   
+    assert np.nanmean(onskyflat_field.data) == pytest.approx(1, abs=1e-2)
+    
+    
+    calibdir = os.path.join(os.path.dirname(__file__), "testcalib")
+    
+    flat_filename = "sim_flatfield_"+str(planet)+"_"+str(band)+".fits"
+    if not os.path.exists(calibdir):
+        os.mkdir(calibdir)
+    onskyflat_field.save(filedir=calibdir, filename=flat_filename)
+    
+    ###### perform flat division
+    # load in the flatfield
+    flat_filepath = os.path.join(calibdir, flat_filename)
+    onsky_flatfield = data.FlatField(flat_filepath)
+
+    
+    flatdivided_dataset = l2a_to_l2b.flat_division(simflat_dataset,onsky_flatfield)
+    print(flatdivided_dataset[0].ext_hdr["HISTORY"])
+    
+    
+	# perform checks after the flat divison for one of the dataset
+    assert(flat_filename in str(flatdivided_dataset[0].ext_hdr["HISTORY"]))
+
+
+	
+	# check the propagated errors for one of the dataset
+    assert flatdivided_dataset[0].err_hdr["Layer_2"] == "FlatField_error"
+    print("mean of all simulated data",np.mean(simflat_dataset.all_data))
+    print("mean of all simulated data error",np.nanmean(simflat_dataset.all_err) )
+    print("mean of all flat divided data:", np.nanmean(flatdivided_dataset.all_data))
+    print("mean of flatfield:", np.nanmean(onsky_flatfield.data))
+    print("mean of flatfield err:", np.nanmean(onsky_flatfield.err))
+    
+    err_flatdiv=np.nanmean(flatdivided_dataset.all_err)
+    err_estimated=np.sqrt(((np.nanmean(onsky_flatfield.data))**2)*(np.nanmean(simflat_dataset.all_err))**2+((np.nanmean(simflat_dataset.all_data))**2)*(np.nanmean(onsky_flatfield.err))**2)
+    print("mean of all flat divided data errors:",err_flatdiv)
+    print("Error estimated:",err_estimated)
+    assert(err_flatdiv == pytest.approx(err_estimated, abs = 1e-2))
+    
+    print(flatdivided_dataset[0].ext_hdr)
+    
+    #creating flatfield using uranus for band4 
```

Comment:
the flat field test for uranus in band4 should be a separate function, since they are independent tests. 1 test per function.